src/communication_managers/iothub_communication_manager/src/iothub_communication_manager.c

```diff
@@ -222,10 +222,11 @@ void IoTHub_CommunicationManager_ConnectionStatus_Callback(
         }
         else
         {
+            int time_until_next_retry = g_next_authentication_attempt_time - now_time;
```

Comment:
The typedef for time_t is likely 64-bits on a modern-day system, and the int type here will cause an overflow as per the [Year 2038 problem](https://en.wikipedia.org/wiki/Year_2038_problem).  ISO C only guarantees time_t to be an arithmetic type--technically it can be floating point on some platforms.  POSIX-compliant platforms guarantee it is an integral type.  You should be able to use the latter (`long int`) and corresponding format specifier (`%ld`).  It is also a `long int` on windows platform if you search [here](https://learn.microsoft.com/en-us/cpp/c-runtime-library/standard-types?view=msvc-170).